scripts/deploy/run-deploy.ts

```diff
@@ -0,0 +1,31 @@
+import { ciDeploy, evergreenDeploy, localDeploy } from "./deploy-production";
+import { red, yellow } from "../utils/colors";
+import { isDryRun } from "./utils/environment";
+import { isOnMainBranch, isWorkingDirectoryClean } from "./utils/git";
+
+const main = async () => {
+  // If this is a dry run, disable the safety checks
+  if (isDryRun) {
+    console.log(yellow("Dry run mode enabled. No changes will be made."));
+  } else {
+    if (!isOnMainBranch()) {
+      console.log(red("You must be on the main branch to deploy!"));
+      process.exit(1);
+    }
+    if (!isWorkingDirectoryClean()) {
+      console.log(red("You must have a clean working directory to deploy"));
+      process.exit(1);
+    }
+  }
+  if (process.argv.includes("--local")) {
+    await localDeploy();
+    return;
+  }
+  if (process.argv.includes("--ci")) {
```

Comment:
Is `isRunningOnCI()` insufficient here?

Reply:
isRunningOnCI checks for the CI env variable that evergreen injects while this checks for the `--ci` flag. I figured it can be an extra safety check that it's both actually a CI build and we are intentionally triggering the CI path. (It also allows us to dry run the CI deploy this way)

Reply:
I see, my personal take is that defining whether we're on CI in 2 different ways is a bit confusing. We could also do a CI dry run by running `CI=true yarn deploy --dry-run`, no? But if this is strongly your preference that's okay with me too!

Reply:
Good point done!